# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Parser.h
# Commit: 1b8c42577e24
# Full Hash: 1b8c42577e2410dbbe050bb30b0ccced9d7c3771
# Author: Ashley Hauck <khyperia@mozilla.com>
# Date: 2019-03-20 10:52:37
# Regressor Bug: 1535166
# File Overlap Count: 1
# Description:
#   Bug 1535166 - Implement computed field names. r=jorendorff
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D23408
# ==============================================================================

diff -r 28370bf14b97 -r 1b8c42577e24 js/src/frontend/Parser.h
--- a/js/src/frontend/Parser.h	Tue Mar 19 22:02:49 2019 +0000
+++ b/js/src/frontend/Parser.h	Tue Mar 19 20:23:03 2019 +0000
@@ -1311,6 +1311,7 @@
                                 HandlePropertyName className,
                                 uint32_t classStartOffset, bool hasHeritage,
                                 size_t& numFieldsWithInitializers,
+                                size_t& numFieldKeys,
                                 ListNodeType& classMembers, bool* done);
   MOZ_MUST_USE bool finishClassConstructor(
       const ParseContext::ClassStatement& classStmt,
@@ -1318,8 +1319,8 @@
       uint32_t classEndOffset, size_t numFieldsWithInitializers,
       ListNodeType& classMembers);
 
-  FunctionNodeType fieldInitializer(YieldHandling yieldHandling,
-                                    HandleAtom atom);
+  FunctionNodeType fieldInitializerOpt(YieldHandling yieldHandling, Node name,
+                                       HandleAtom atom, size_t& numFieldKeys);
   FunctionNodeType synthesizeConstructor(HandleAtom className,
                                          uint32_t classNameOffset);
 